### PR TITLE
feat: transactions

### PR DESCRIPTION
**Pull request by douglasmdev**
## Description

- Treats batch (executemany)
- converts errors to sqlite3, e.g. IntegrityError
- Statement parsing with is_dml 
- decorators for  check_connection, check_cursor, check_thread and such
- commit(), rollback() e _begin_transaction
- _in_transaction stemming from current command/statement (_update_in_transaction)
- line convertion on demand in __next__

### CODE REVIEW

**Comment by douglasmdev** [comment]
`poetry run pytest tests/dbapi2/test_transactions.py`

Log: 
[test_transaction.log](https://github.com/profusion/libsql-client-py/files/11488367/test_transaction.log)

**Comment by douglasmdev** [comment]
Log:
[test_transaction.log](https://github.com/profusion/libsql-client-py/files/11488606/test_transaction.log)

**Comment by douglasmdev** [comment]
[test_transaction.log](https://github.com/profusion/libsql-client-py/files/11489168/test_transaction.log)

File: libsql_client/dbapi2/_hrana.py [hunk 1]
```diff
@@ -190,6 +192,7 @@ def __init__(
         self._session = self._acquire_session()
         self._conn = self._acquire_connection(database)
         self._stream = self._create_stream()
+        self.autocommit = kwargs.pop("autocommit", LEGACY_TRANSACTION_CONTROL)
```
Comment (barbieri) [inline libsql_client/dbapi2/_hrana.py:195]: this should be in `_types:Connection`, since all should support it.

Comment (barbieri) [reply libsql_client/dbapi2/_hrana.py:195]: btw, you can declare as the C-counterpart as a init argument:

```
autocommit: AutoCommit = AutoCommit.AUTO_COMMIT_LEGACY
```

btw, their enum is `Autocommit` and not `AutoCommit`, adjust that one as well

File: libsql_client/dbapi2/_types.py [hunk 2]
```diff
@@ -37,11 +37,13 @@
     ProgrammingError,
 )
 from sqlite3.dump import _iterdump as sqlite3_iterdump  # type: ignore
+
 from ..hrana import proto
 from ..hrana.convert import _value_from_proto
 from ..client import LibsqlError
 
-from ._utils import log_obj, iter_sql_statements
+from ._utils import log_obj, iter_sql_statements, LEGACY_TRANSACTION_CONTROL
```
Comment (barbieri) [inline libsql_client/dbapi2/_types.py:45]: get it from sqlite3.dbapi2

File: libsql_client/dbapi2/_hrana.py (hunk 1 above)
Comment (barbieri) [reply libsql_client/dbapi2/_hrana.py:195]: actually, it should be `autocommit: Union[bool, LEGACY_TRANSACTION_CONTROL] = LEGACY_TRANSACTION_CONTROL ` just like property.

File: libsql_client/dbapi2/_types.py (hunk 2 above)
Comment (douglasmdev) [reply libsql_client/dbapi2/_types.py:45]: There's no `LEGACY_TRANSACTION_CONTROL` there, not in sqlite3 even. It Might be an internal setting.

Comment (barbieri) [reply libsql_client/dbapi2/_types.py:45]: ok

File: libsql_client/dbapi2/_hrana.py (hunk 1 above)
Comment (douglasmdev) [reply libsql_client/dbapi2/_hrana.py]: `LEGACY_TRANSACTION_CONTROL` is not a valid type. Did you mean 

`Union[bool, Autocommit]`?

**Pull request closed without merging**